multimodal/src/autogluon/multimodal/utils/onnx.py

```diff
@@ -68,7 +92,21 @@ def __init__(self, model, providers: Optional[Union[dict, List[str]]] = None):
                 ),
                 ("CPUExecutionProvider", {}),
             ]
-        self.sess = ort.InferenceSession(model.SerializeToString(), providers=providers)
+
+        if len(providers) == 1 and get_provider_name(providers[0]) == "TensorrtExecutionProvider":
+            if not tensorrt_imported:
+                raise ImportError(
+                    "tensorrt package is not installed. The package can be install via `pip install tensorrt`."
+                )
+
+        self.sess = ort.InferenceSession(onnx_model.SerializeToString(), providers=providers)
```

Comment:
The `provider` argument of `OnnxModule` should be easy-to-remember strings, e.g., `tensorrt_gpu`, `onnx_gpu`, `tensorrt_cpu`, `onnx_cpu`? Then we can map these strings to more complex providers used by `ort.InferenceSession`.

Reply:
1. There wouldn't exist a `tensorrt_cpu`, since tensorrt would only serve the purpose of nvidia GPU.
2. `onnx_gpu` for OnnxModule: AMD also provider [ROCm Execution Provider](https://onnxruntime.ai/docs/execution-providers/ROCm-ExecutionProvider.html), which would make `onnx_gpu` argument for OnnxModule confusing.

Reply:
I would say there is a trade-off between easy-to-rememberr constants vs transparency to onnxruntime config.